pkg/ibmcsidriver/controller.go

```diff
@@ -566,30 +591,167 @@ func (csiCS *CSIControllerServer) GetCapacity(ctx context.Context, req *csi.GetC
 func (csiCS *CSIControllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
 	ctxLogger, requestID := utils.GetContextLogger(ctx, false)
 	// populate requestID in the context
-	_ = context.WithValue(ctx, provider.RequestID, requestID)
+	ctx = context.WithValue(ctx, provider.RequestID, requestID)
+	ctxLogger.Info("CSIControllerServer-CreateSnapshot... ", zap.Reflect("Request", req))
+	defer metrics.UpdateDurationFromStart(ctxLogger, "CreateSnapshot", time.Now())
+
+	//Feature flag to enable/disable CreateSnapshot feature.
+	if strings.ToLower(os.Getenv("IS_SNAPSHOT_ENABLED")) == "false" {
+		ctxLogger.Warn("CreateSnapshot functionality is disabled.")
+		time.Sleep(10 * time.Minute) //To avoid multiple retries from kubernetes to CSI Driver
```

Comment:
can you check if this is really required even we are returning MethodUnimplemented